adapters/x509_openssl.c

```diff
@@ -139,8 +141,21 @@ static int load_key_RSA(SSL_CTX* ssl_ctx, EVP_PKEY* evp_key)
 {
     int result;
     /*Codes_SRS_X509_OPENSSL_02_005: [ x509_openssl_add_credentials shall use PEM_read_bio_RSAPrivateKey to read the x509 private key. ]*/
-    RSA* privatekey = EVP_PKEY_get1_RSA(evp_key);
-    if (privatekey == NULL)
+    RSA *privatekey;
+#ifdef defined(OPENSSL_VERSION_NUMBER) AND (OPENSSL_VERSION_NUMBER >= OPENSSL_VERSION_3)
+    if (evp_key->type == EVP_PKEY_RSA || evp_key->type == EVP_PKEY_RSA_PSS)
+    {
+        privatekey = NULL;
+
+        if (privatekey != NULL)
```

Comment:
what does this do? looks like privatekey is null.

Reply:
Sorry, I'm making changes to the file, I'll ping you once it's ready for review.